test/acceptance/test/pages/add_application_page.go

```diff
@@ -0,0 +1,34 @@
+package pages
+
+import (
+	"github.com/sclevine/agouti"
+)
+
+type AddAppPageElements struct {
+	AddAppHeader         *agouti.Selection
+	AppName              *agouti.Selection
+	AppNamespace         *agouti.Selection
+	AppRepoURL           *agouti.Selection
+	ConfigRepoURL        *agouti.Selection
+	PathToManifests      *agouti.Selection
+	Branch               *agouti.Selection
+	AutoMergeCheck       *agouti.Selection
+	SubmitButton         *agouti.Selection
+	AuthenticationButton *agouti.Selection
+}
+
+func GetAddAppPageElements(webDriver *agouti.Page) *AddAppPageElements {
+	addApplicationPage := AddAppPageElements{
+		AddAppHeader:         webDriver.FindByXPath(`//*[@id="app"]//nav//h2[text()="Add Application"]`),
+		AppName:              webDriver.FindByID("name"),
+		AppNamespace:         webDriver.FindByID("namespace"),
+		AppRepoURL:           webDriver.FindByID("url"),
+		ConfigRepoURL:        webDriver.FindByID("configUrl"),
+		PathToManifests:      webDriver.FindByID("path"),
+		Branch:               webDriver.FindByID("branch"),
+		AutoMergeCheck:       webDriver.FindByXPath(`//input[@type="checkbox" and @class="jss12 MuiSwitch-input"]`),
```

Comment:
Our class names are programatically generated and non-deterministic (I don't know the exact algo that `styled-components` uses to come up with them`).

This `jss12` class is not something that a human entered and will most likely change on the next build of the UI. `.MuiSwitch-input` **should** stay the same, but since this is an `and` this will not be a good way to test.

Reply:
@jpellizzari I don't _need_ to use the class name since there's only 1 checkbox in the whole page. I could skip the classname altogther and it would suffice OR I could use other xpath methods like `contains()`. I'll make the change.

Reply:
That works better. I would still like to have a chat about this approach to UI testing.